lib/improver/blending/weights.py

```diff
@@ -637,6 +636,52 @@ def _calculate_weights(self, cube, weights_cube=None):
 
         return new_weights_cube
 
+    def _create_slice_iter(self, cube):
+        """Returns an iterator over which to slice each input cube"""
+        if cube.coord_dims(self.weighting_coord_name):
+            slice_iterator = [
+                cube.coord(self.weighting_coord_name),
+                cube.coord(axis='y'), cube.coord(axis='x')]
+        else:
+            slice_iterator = [cube.coord(axis='y'), cube.coord(axis='x')]
+        return slice_iterator
+
+    def _slice_input_cubes(self, cubes):
```

Comment:
The plugin was written to take a list of unmerged cubes, whereas the CLI needs to merge the cubes first in order to do things like ordering by model id (which we need to do so that the order of the weights array / list matches the merged cube passed into `WeightedBlendAcrossWholeDimension`...).  If we pass a merged cubelist into this plugin (from different models when we're blending across the model id coordinate), we have to un-merge it in order to calculate the weights.  This is a horrible hack but was better in my opinion than rewriting the plugin at this stage.  Some of the tidying up (I hope) should be covered by the work to align weights calculation plugin return values in #657 .

Reply:
Ok, can you write some of this information on that ticket if you haven't already